File: jump/tammes_7.jl
# NLP written by GAMS Convert at 05/15/24 11:33:41
#
# Equation counts
#     Total        E        G        L        N        X        C        B
#        28        7        0       21        0        0        0        0
#
# Variable counts
#                  x        b        i      s1s      s2s       sc       si
#     Total     cont   binary  integer     sos1     sos2    scont     sint
#        22       22        0        0        0        0        0        0
# FX      0
#
# Nonzero counts
#     Total    const       NL
#       168       21      147
#
# Reformulation has removed 1 variable and 1 equation

using JuMP
using MathOptInterface
const MOI = MathOptInterface

model = m = Model()

@variable(m, -1 <= x1 <= 1, start=0)
@variable(m, -1 <= x2 <= 1, start=0)
@variable(m, -1 <= x3 <= 1, start=0)
@variable(m, -1 <= x4 <= 1, start=0)
@variable(m, -1 <= x5 <= 1, start=0)
@variable(m, -1 <= x6 <= 1, start=0)
@variable(m, -1 <= x7 <= 1, start=0)
@variable(m, -1 <= x8 <= 1, start=0)
@variable(m, -1 <= x9 <= 1, start=0)
@variable(m, -1 <= x10 <= 1, start=0)
@variable(m, -1 <= x11 <= 1, start=0)
@variable(m, -1 <= x12 <= 1, start=0)
@variable(m, -1 <= x13 <= 1, start=0)
@variable(m, -1 <= x14 <= 1, start=0)
@variable(m, -1 <= x15 <= 1, start=0)
@variable(m, -1 <= x16 <= 1, start=0)
@variable(m, -1 <= x17 <= 1, start=0)
@variable(m, -1 <= x18 <= 1, start=0)
@variable(m, -1 <= x19 <= 1, start=0)
@variable(m, -1 <= x20 <= 1, start=0)
@variable(m, -1 <= x21 <= 1, start=0)
@variable(m, x22, start=0)

@objective(m, Max, x22)

@NLconstraint(m, e1, x1^2 + x8^2 + x15^2 == 1)
@NLconstraint(m, e2, x2^2 + x9^2 + x16^2 == 1)
@NLconstraint(m, e3, x3^2 + x10^2 + x17^2 == 1)
@NLconstraint(m, e4, x4^2 + x11^2 + x18^2 == 1)
@NLconstraint(m, e5, x5^2 + x12^2 + x19^2 == 1)
@NLconstraint(m, e6, x6^2 + x13^2 + x20^2 == 1)
@NLconstraint(m, e7, x7^2 + x14^2 + x21^2 == 1)
@NLconstraint(m, e8, -sqrt((x1 - x2)^2 + (x8 - x9)^2 + (x15 - x16)^2) + x22
    <= 0)
@NLconstraint(m, e9, -sqrt((x1 - x3)^2 + (x8 - x10)^2 + (x15 - x17)^2) + x22
    <= 0)
@NLconstraint(m, e10, -sqrt((x1 - x4)^2 + (x8 - x11)^2 + (x15 - x18)^2) + x22
    <= 0)
@NLconstraint(m, e11, -sqrt((x1 - x5)^2 + (x8 - x12)^2 + (x15 - x19)^2) + x22
    <= 0)
@NLconstraint(m, e12, -sqrt((x1 - x6)^2 + (x8 - x13)^2 + (x15 - x20)^2) + x22
    <= 0)
@NLconstraint(m, e13, -sqrt((x1 - x7)^2 + (x8 - x14)^2 + (x15 - x21)^2) + x22
    <= 0)
@NLconstraint(m, e14, -sqrt((x2 - x3)^2 + (x9 - x10)^2 + (x16 - x17)^2) + x22
    <= 0)
@NLconstraint(m, e15, -sqrt((x2 - x4)^2 + (x9 - x11)^2 + (x16 - x18)^2) + x22
    <= 0)
@NLconstraint(m, e16, -sqrt((x2 - x5)^2 + (x9 - x12)^2 + (x16 - x19)^2) + x22
    <= 0)
@NLconstraint(m, e17, -sqrt((x2 - x6)^2 + (x9 - x13)^2 + (x16 - x20)^2) + x22
    <= 0)
@NLconstraint(m, e18, -sqrt((x2 - x7)^2 + (x9 - x14)^2 + (x16 - x21)^2) + x22
    <= 0)
@NLconstraint(m, e19, -sqrt((x3 - x4)^2 + (x10 - x11)^2 + (x17 - x18)^2) + x22
    <= 0)
@NLconstraint(m, e20, -sqrt((x3 - x5)^2 + (x10 - x12)^2 + (x17 - x19)^2) + x22
    <= 0)
@NLconstraint(m, e21, -sqrt((x3 - x6)^2 + (x10 - x13)^2 + (x17 - x20)^2) + x22
    <= 0)
@NLconstraint(m, e22, -sqrt((x3 - x7)^2 + (x10 - x14)^2 + (x17 - x21)^2) + x22
    <= 0)
@NLconstraint(m, e23, -sqrt((x4 - x5)^2 + (x11 - x12)^2 + (x18 - x19)^2) + x22
    <= 0)
@NLconstraint(m, e24, -sqrt((x4 - x6)^2 + (x11 - x13)^2 + (x18 - x20)^2) + x22
    <= 0)
@NLconstraint(m, e25, -sqrt((x4 - x7)^2 + (x11 - x14)^2 + (x18 - x21)^2) + x22
    <= 0)
@NLconstraint(m, e26, -sqrt((x5 - x6)^2 + (x12 - x13)^2 + (x19 - x20)^2) + x22
    <= 0)
@NLconstraint(m, e27, -sqrt((x5 - x7)^2 + (x12 - x14)^2 + (x19 - x21)^2) + x22
    <= 0)
@NLconstraint(m, e28, -sqrt((x6 - x7)^2 + (x13 - x14)^2 + (x20 - x21)^2) + x22
    <= 0)

File: pyomo/tammes_7.py
# NLP written by GAMS Convert at 05/15/24 11:46:43
#
# Equation counts
#     Total        E        G        L        N        X        C        B
#        28        7        0       21        0        0        0        0
#
# Variable counts
#                  x        b        i      s1s      s2s       sc       si
#     Total     cont   binary  integer     sos1     sos2    scont     sint
#        22       22        0        0        0        0        0        0
# FX      0
#
# Nonzero counts
#     Total    const       NL
#       168       21      147
#
# Reformulation has removed 1 variable and 1 equation

from pyomo.environ import *

model = m = ConcreteModel()

m.x1 = Var(within=Reals, bounds=(-1,1), initialize=0)
m.x2 = Var(within=Reals, bounds=(-1,1), initialize=0)
m.x3 = Var(within=Reals, bounds=(-1,1), initialize=0)
m.x4 = Var(within=Reals, bounds=(-1,1), initialize=0)
m.x5 = Var(within=Reals, bounds=(-1,1), initialize=0)
m.x6 = Var(within=Reals, bounds=(-1,1), initialize=0)
m.x7 = Var(within=Reals, bounds=(-1,1), initialize=0)
m.x8 = Var(within=Reals, bounds=(-1,1), initialize=0)
m.x9 = Var(within=Reals, bounds=(-1,1), initialize=0)
m.x10 = Var(within=Reals, bounds=(-1,1), initialize=0)
m.x11 = Var(within=Reals, bounds=(-1,1), initialize=0)
m.x12 = Var(within=Reals, bounds=(-1,1), initialize=0)
m.x13 = Var(within=Reals, bounds=(-1,1), initialize=0)
m.x14 = Var(within=Reals, bounds=(-1,1), initialize=0)
m.x15 = Var(within=Reals, bounds=(-1,1), initialize=0)
m.x16 = Var(within=Reals, bounds=(-1,1), initialize=0)
m.x17 = Var(within=Reals, bounds=(-1,1), initialize=0)
m.x18 = Var(within=Reals, bounds=(-1,1), initialize=0)
m.x19 = Var(within=Reals, bounds=(-1,1), initialize=0)
m.x20 = Var(within=Reals, bounds=(-1,1), initialize=0)
m.x21 = Var(within=Reals, bounds=(-1,1), initialize=0)
m.x22 = Var(within=Reals, bounds=(None,None), initialize=0)

m.obj = Objective(sense=maximize, expr= m.x22)

m.e1 = Constraint(expr= m.x1**2 + m.x8**2 + m.x15**2 == 1)
m.e2 = Constraint(expr= m.x2**2 + m.x9**2 + m.x16**2 == 1)
m.e3 = Constraint(expr= m.x3**2 + m.x10**2 + m.x17**2 == 1)
m.e4 = Constraint(expr= m.x4**2 + m.x11**2 + m.x18**2 == 1)
m.e5 = Constraint(expr= m.x5**2 + m.x12**2 + m.x19**2 == 1)
m.e6 = Constraint(expr= m.x6**2 + m.x13**2 + m.x20**2 == 1)
m.e7 = Constraint(expr= m.x7**2 + m.x14**2 + m.x21**2 == 1)
m.e8 = Constraint(expr= -sqrt((m.x1 - m.x2)**2 + (m.x8 - m.x9)**2 + (m.x15 -
    m.x16)**2) + m.x22 <= 0)
m.e9 = Constraint(expr= -sqrt((m.x1 - m.x3)**2 + (m.x8 - m.x10)**2 + (m.x15 -
    m.x17)**2) + m.x22 <= 0)
m.e10 = Constraint(expr= -sqrt((m.x1 - m.x4)**2 + (m.x8 - m.x11)**2 + (m.x15 -
    m.x18)**2) + m.x22 <= 0)
m.e11 = Constraint(expr= -sqrt((m.x1 - m.x5)**2 + (m.x8 - m.x12)**2 + (m.x15 -
    m.x19)**2) + m.x22 <= 0)
m.e12 = Constraint(expr= -sqrt((m.x1 - m.x6)**2 + (m.x8 - m.x13)**2 + (m.x15 -
    m.x20)**2) + m.x22 <= 0)
m.e13 = Constraint(expr= -sqrt((m.x1 - m.x7)**2 + (m.x8 - m.x14)**2 + (m.x15 -
    m.x21)**2) + m.x22 <= 0)
m.e14 = Constraint(expr= -sqrt((m.x2 - m.x3)**2 + (m.x9 - m.x10)**2 + (m.x16 -
    m.x17)**2) + m.x22 <= 0)
m.e15 = Constraint(expr= -sqrt((m.x2 - m.x4)**2 + (m.x9 - m.x11)**2 + (m.x16 -
    m.x18)**2) + m.x22 <= 0)
m.e16 = Constraint(expr= -sqrt((m.x2 - m.x5)**2 + (m.x9 - m.x12)**2 + (m.x16 -
    m.x19)**2) + m.x22 <= 0)
m.e17 = Constraint(expr= -sqrt((m.x2 - m.x6)**2 + (m.x9 - m.x13)**2 + (m.x16 -
    m.x20)**2) + m.x22 <= 0)
m.e18 = Constraint(expr= -sqrt((m.x2 - m.x7)**2 + (m.x9 - m.x14)**2 + (m.x16 -
    m.x21)**2) + m.x22 <= 0)
m.e19 = Constraint(expr= -sqrt((m.x3 - m.x4)**2 + (m.x10 - m.x11)**2 + (m.x17
    - m.x18)**2) + m.x22 <= 0)
m.e20 = Constraint(expr= -sqrt((m.x3 - m.x5)**2 + (m.x10 - m.x12)**2 + (m.x17
    - m.x19)**2) + m.x22 <= 0)
m.e21 = Constraint(expr= -sqrt((m.x3 - m.x6)**2 + (m.x10 - m.x13)**2 + (m.x17
    - m.x20)**2) + m.x22 <= 0)
m.e22 = Constraint(expr= -sqrt((m.x3 - m.x7)**2 + (m.x10 - m.x14)**2 + (m.x17
    - m.x21)**2) + m.x22 <= 0)
m.e23 = Constraint(expr= -sqrt((m.x4 - m.x5)**2 + (m.x11 - m.x12)**2 + (m.x18
    - m.x19)**2) + m.x22 <= 0)
m.e24 = Constraint(expr= -sqrt((m.x4 - m.x6)**2 + (m.x11 - m.x13)**2 + (m.x18
    - m.x20)**2) + m.x22 <= 0)
m.e25 = Constraint(expr= -sqrt((m.x4 - m.x7)**2 + (m.x11 - m.x14)**2 + (m.x18
    - m.x21)**2) + m.x22 <= 0)
m.e26 = Constraint(expr= -sqrt((m.x5 - m.x6)**2 + (m.x12 - m.x13)**2 + (m.x19
    - m.x20)**2) + m.x22 <= 0)
m.e27 = Constraint(expr= -sqrt((m.x5 - m.x7)**2 + (m.x12 - m.x14)**2 + (m.x19
    - m.x21)**2) + m.x22 <= 0)
m.e28 = Constraint(expr= -sqrt((m.x6 - m.x7)**2 + (m.x13 - m.x14)**2 + (m.x20
    - m.x21)**2) + m.x22 <= 0)
